### PR TITLE
Fix/nonce-ssg

### PR DESCRIPTION
**Pull request by vejja**
Fix the issue where an immutable nonce value is injected in all pre-rendered pages if the `nonce` option is set to `true`

## Types of changes

- [x] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

This PR solves the issue

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes (if not applicable, please state why)

### CODE REVIEW

**Comment by vejja** [comment]
Step 1 : add incremental test

**Comment by vejja** [comment]
step 2 : modify code blocks

**Comment by vejja** [comment]
step 3 - cleanup
Ready for review

**Comment by Baroshem** [comment]
This PR looks good. I will review it tomorrow probably but would also love the feedback from @trijpstra-fourlights :)

**Comment by Baroshem** [comment]
Hey @vejja 

Lets reschedule merging this PR to rc.1 to rc.2 instead. 

It might be safer to do that and focus on releasing today the version 1 which I have already tested.

Dont worry though! Your PR is first to be merged to 1.0.0-rc.2 branch so it wont dissapear :)

**Review by trijpstra-fourlights — APPROVED** [review]
Nice work, one small nitpick but LGTM

File: playground/nuxt.config.ts [hunk 1]
```diff
@@ -23,6 +23,7 @@ export default defineNuxtConfig({
     },
     rateLimiter: {
       tokensPerInterval: 10
-    }
+    },
+    nonce: true
```
[the remark this answers is not in the document]
Comment (vejja) [reply playground/nuxt.config.ts:27]: Correct, should be removed

[the remark this answers is not in the document]
Comment (Baroshem) [reply playground/nuxt.config.ts:27]: Awesome guys!

Let's keep it in mind for 1.0.0-rc.2 version.

Thanks for the amazing work! 💚

**Comment by Baroshem** [comment]
Update.

Let's make it ready for the 1.0.0-rc.3. I needed to push a fix to 1.0.0-rc.1 because there was a bug that I didn't catch during testing. It was triggered once released to NPM :(

**Review by Baroshem — APPROVED** [review]

**Pull request merged**